io/io/v7/src/TFile.cxx

```diff
@@ -191,7 +191,7 @@ std::string ROOT::Experimental::TFile::SetCacheDir(std::string_view path) {
   std::lock_guard<std::mutex> lock(GetCacheDirMutex());
 
   std::string ret = ::TFile::GetCacheFileDir();
-  ::TFile::SetCacheFileDir(path.to_string().c_str());
+  ::TFile::SetCacheFileDir(path.data());
```

Comment:
At the end I didn't like idea of overload. It would cause more lines to be changed at this point and wouldn't work with `TString` (ambiguity). Let's do more minimal change for now.